gwpy/types/tests/test_array2d.py

```diff
@@ -216,6 +216,16 @@ def test_single_column_slice(self):
         utils.assert_array_equal(b.xindex, a.xindex[0:1])
         utils.assert_array_equal(b.yindex, a.yindex)
 
+    def test_two_index_arrays(self):
+        """Test that subsetting with two index arrays works correctly."""
+        # create an array with indices
+        rawa = numpy.arange(12).reshape((4, 3))
+        a = Array2D(rawa)
+        exp = numpy.array([3, 11])
+        ind1, ind2 = numpy.array([1, 3]), numpy.array([0, 2])
+        assert numpy.all(rawa[ind1, ind2] == exp)
+        assert numpy.all(a[ind1, ind2].value == exp)
```

Comment:
I think it's better here to use `numpy.testing.assert_array_equal` (with the appropriate extra `import` statement at the top of the module):

```suggestion
        assert assert_array_equal(rawa[ind1, ind2], exp)
        assert assert_array_equal(a[ind1, ind2].value, exp)
```